### PR TITLE
feat: added the top_k param for voting parallel

### PR DESCRIPTION
**Pull request by ocworld**
feat: added the top_k param for voting parallel

### CODE REVIEW

File: src/main/scala/com/microsoft/ml/spark/lightgbm/LightGBMParams.scala [hunk 1]
```diff
@@ -17,6 +17,15 @@ trait LightGBMExecutionParams extends Wrappable {
   def getParallelism: String = $(parallelism)
   def setParallelism(value: String): this.type = set(parallelism, value)
 
+  val topK = new Param[Int](this, "topK",
+    "The top_k value used in Voting parallel, " +
+      "set this to larger value for more accurate result, but it will slow down the training speed. " +
+      "It should be greater than 0")
+  setDefault(topK -> LightGBMConstants.DefaultTopK)
```
Comment (imatiach-msft) [inline src/main/scala/com/microsoft/ml/spark/lightgbm/LightGBMParams.scala]: interesting idea to put these in constants, I like it

**Comment by imatiach-msft** [comment]
[0m[[0m[0minfo[0m] [0m[0m[32mVerifyLightGBMClassifier:[0m[0m
[0m[[0m[0minfo[0m] [0m[0m[31m- Serialization Fuzzing *** FAILED ***[0m[0m
[0m[[0m[0minfo[0m] [0m[0m[31m  scala.NotImplementedError: The default jsonEncode only supports string, vector and matrix. org.apache.spark.ml.param.Param must override jsonEncode for java.lang.Integer.[0m[0m
[0m[[0m[0minfo[0m] [0m[0m[31m  at org.apache.spark.ml.param.Param.jsonEncode(params.scala:100)[0m[0m
[0m[[0m[0minfo[0m] [0m[0m[31m  at org.apache.spark.ml.util.DefaultParamsWriter$$anonfun$6.apply(ReadWrite.scala:463)[0m[0m
[0m[[0m[0minfo[0m] [0m[0m[31m  at org.apache.spark.ml.util.DefaultParamsWriter$$anonfun$6.apply(ReadWrite.scala:462)[0m[0m
[0m[[0m[0minfo[0m] [0m[0m[31m  at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)[0m[0m
[0m[[0m[0minfo[0m] [0m[0m[31m  at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)[0m[0m
[0m[[0m[0minfo[0m] [0m[0m[31m  at scala.collection.mutable.ResizableArray$class.foreach(ResizableArray.scala:59)[0m[0m
[0m[[0m[0minfo[0m] [0m[0m[31m  at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:48)[0m[0m
[0m[[0m[0minfo[0m] [0m[0m[31m  at scala.collection.TraversableLike$class.map(TraversableLike.scala:234)[0m[0m
[0m[[0m[0minfo[0m] [0m[0m[31m  at scala.collection.AbstractTraversable.map(Traversable.scala:104)[0m[0m
[0m[[0m[0minfo[0m] [0m[0m[31m  at org.apache.spark.ml.util.DefaultParamsWriter$.getMetadataToSave(ReadWrite.scala:462)[0m[0m
[0m[[0m[0minfo[0m] [0m[0m[31m  ...[0m[0m

File: src/main/scala/com/microsoft/ml/spark/lightgbm/LightGBMParams.scala [hunk 2]
```diff
@@ -17,6 +17,15 @@ trait LightGBMExecutionParams extends Wrappable {
   def getParallelism: String = $(parallelism)
   def setParallelism(value: String): this.type = set(parallelism, value)
 
+  val topK = new Param[Int](this, "topK",
```
Comment (imatiach-msft) [inline src/main/scala/com/microsoft/ml/spark/lightgbm/LightGBMParams.scala]: you need to use IntParam instead, see other params

**Comment by imatiach-msft** [comment]
added comment.  to fix the test you just need to use our IntParam instead of Param[Int]

File: src/main/scala/com/microsoft/ml/spark/lightgbm/LightGBMParams.scala (hunk 1 above)
Comment (ocworld) [reply src/main/scala/com/microsoft/ml/spark/lightgbm/LightGBMParams.scala]: Thanks for your review

File: src/main/scala/com/microsoft/ml/spark/lightgbm/LightGBMParams.scala (hunk 2 above)
Comment (ocworld) [reply src/main/scala/com/microsoft/ml/spark/lightgbm/LightGBMParams.scala]: I've fixed it.

**Comment by ocworld** [comment]
Can I see logs about failure in UnitTests cognitive?

**Comment by mhamilton723** [comment]
@ocworld none of these build issues were your faulty. Will fix these now and then push in your changes :)

**Comment by ocworld** [comment]
@ahnlab-oss

**Pull request merged**